dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -15,266 +15,181 @@
-     * @param entry entry to check permissions for
+     * Check if a the ID of a user matches the specified ID.
+     * @param user user to be checked, null if user not logged in
+     * @param userId id to match
      */
-    default void checkUserOwnsEntry(User user, Entry entry) {
-        if (entry.getUsers().stream().noneMatch(u -> ((User)(u)).getId() == user.getId())) {
-            throw new CustomWebApplicationException("Forbidden: you do not have the credentials required to access this entry.",
-                    HttpStatus.SC_FORBIDDEN);
-        }
-    }
-
-    static boolean userCannotRead(User user, Entry entry) {
-        return !user.getIsAdmin() && (entry.getUsers()).stream().noneMatch(u -> ((User)(u)).getId() == user.getId());
+    default void checkUserId(User user, long userId) {
+        throwIf(user == null || user.getId() != userId, FORBIDDEN_ID_MISMATCH_MESSAGE, HttpStatus.SC_FORBIDDEN);
     }
 
     /**
-     * Checks if a user can read an entry. Default implementation
-     * is to invoke <code>checkUser</code>. Implmenations that support
-     * more nuanched sharing should override.
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for()
+     * Check if a specified entry exists (is not null).
+     * @param entry entry to be checked
      */
-    default void checkUserCanRead(User user, Entry entry) {
-        checkUser(user, entry);
+    default void checkExistsEntry(Entry<?, ?> entry) {
+        throwIf(entry == null, "Entry not found.", HttpStatus.SC_NOT_FOUND);
     }
 
     /**
-     * Checks if a user can modify an entry. Default implementation
-     * is to invoke <code>checkUser</code>. Implementations that support
-     * more nuanced sharing should override.
-     *
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for()
+     * Check if a specified user exists (is not null).
+     * @param user user to be checked
      */
-    default void checkUserCanUpdate(User user, Entry entry) {
-        checkUserOwnsEntry(user, entry);
+    default void checkExistsUser(User user) {
+        throwIf(user == null, "User not found.", HttpStatus.SC_NOT_FOUND);
     }
 
     /**
-     * Checks if a user can delete an entry. Default implementation
-     * is to invoke <code>checkUser</code>. Implementations that support
-     * more nuanced sharing should override.
-     *
-     * @param user the user that is requesting something
-     * @param entry entry to check permissions for()
+     * Check if a specified organization exists (is not null).
+     * @param organization organization to be checked
      */
-    default void checkUserCanDelete(User user, Entry entry) {
-        checkUser(user, entry);
+    default void checkExistsOrganization(Organization organization) {
+        throwIf(organization == null, "Organization not found.", HttpStatus.SC_NOT_FOUND);
```

Comment:
We also have [this method](https://github.com/dockstore/dockstore/blob/bdfecd28a9668fa4aee8188ac49a71648144998a/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java#L239). Can they be combined in this specific case, and more generically, I'm wondering if all these different null checks belong in AuthenticatedResourceInterface (I'm not sure).

Reply:
Removed `checkExistsOrganization`.

Reply:
I moved some of the null checks to the particular classes where they are used.